src/node/transaction.cpp

```diff
@@ -30,16 +30,15 @@ TransactionError BroadcastTransaction(const CTransactionRef tx, uint256& hashTx,
     if (!fHaveMempool && !fHaveChain) {
         // push to local node and sync with wallets
         TxValidationState state;
-        bool fMissingInputs;
-        if (!AcceptToMemoryPool(mempool, state, std::move(tx), &fMissingInputs,
+        if (!AcceptToMemoryPool(mempool, state, std::move(tx),
                                 nullptr /* plTxnReplaced */, false /* bypass_limits */, highfee)) {
             if (state.IsInvalid()) {
+                if (state.GetResult() == TxValidationResult::TX_MISSING_INPUTS) {
```

Comment:
Good catch.

Reply:
It was intentional to maintain existing behaviour, but you're right that it looks odd and buggy to anyone reading the code now. Rather than write a comment explaining that I'm maintaining existing behaviour, I'm just going to set `err_string` in all cases.